lib/LaTeXML/Core/Tokens.pm

```diff
@@ -84,13 +100,24 @@ sub beDigested {
 
 sub neutralize {
   my ($self, @extraspecials) = @_;
-  return Tokens(map { $_->neutralize(@extraspecials) } $self->unlist); }
+  my @neutralized = map { $_->without_dont_expand } $self->unlist;
+  if (ref @$self[-1] eq 'ARRAY') {    # preserve marks
+    push(@neutralized, @$self[-1]); }
```

Comment:
The middle is dead code, here.  What is the expectation when both a no-expand-once (from `\the`) and dont-expand (from `\noexpand`) are present?  Or can that even happen?

Reply:
The current code does not allow for an arrayref mask made via `\the` to enter `Tokens` as a regular `Token`. It always slots that information as a mask in this magical last position of the array. Which is also why the middle isn't dead code - the call to `->unlist` ignores the masks, and then the check on `[-1]` reintroduces them in the new `@neutralized` array. 

Meanwhile, the `\noexpand` neutralization is still done in the same way, via a special `Token` that has `$_[2]` set.

Reply:
This code may get a lot cleaner if I create a new object inheriting from `Tokens`, say `MaskedTokens`, which does all of these acrobatics internally. It will be marginally slower, since objects aren't free, but will likely clean up the logic a lot.

Reply:
Sure, but you never use `@neutralized`, which is why it's dead.

Reply:
Ah oops! This is a bug actually, I think I mangled that unintentionally... Might not be well-tested in the testing suite too.

Reply:
I added a patch, should be accurate to my intentions now.